cg/meta/transfer/external_data.py

```diff
@@ -5,210 +5,194 @@
 from housekeeper.store.models import Version
 
 from cg.apps.slurm.slurm_api import SlurmAPI
-from cg.constants import HK_FASTQ_TAGS
+from cg.constants import HK_FASTQ_TAGS, FileExtensions
+from cg.constants.constants import CaseActions
 from cg.meta.meta import MetaAPI
 from cg.meta.rsync.sbatch import ERROR_RSYNC_FUNCTION, RSYNC_CONTENTS_COMMAND
+from cg.meta.transfer.utils import are_all_fastq_valid
 from cg.models.cg_config import CGConfig
 from cg.models.slurm.sbatch import Sbatch
 from cg.store.models import Case, Customer, Sample
-from cg.utils.checksum.checksum import check_md5sum, extract_md5sum
+from cg.utils.files import get_files_matching_pattern
 
 LOG = logging.getLogger(__name__)
 
 
 class ExternalDataAPI(MetaAPI):
     def __init__(self, config: CGConfig):
         super().__init__(config)
-        self.destination_path: str = config.external.hasta
-        self.source_path: str = config.external.caesar
-        self.base_path: str = config.data_delivery.base_path
         self.account: str = config.data_delivery.account
+        self.base_path: str = config.data_delivery.base_path
+        self.customer_id: str | None = None
+        self.destination_path: str = config.external.hasta
+        self.dry_run: bool | None = None
+        self.force: bool | None = None
         self.mail_user: str = config.data_delivery.mail_user
-        self.slurm_api: SlurmAPI = SlurmAPI()
         self.RSYNC_FILE_POSTFIX: str = "_rsync_external_data"
+        self.slurm_api: SlurmAPI = SlurmAPI()
+        self.source_path: str = config.external.caesar
+        self.ticket: str | None = None
+
+    def _set_parameters(self, ticket: str, dry_run: bool, force: bool | None = False):
```

Comment:
Although we agreed that we should not use setters, I think this is more of a wrapper for assigning values to a lot of attributes that is more DRY

Reply:
Can't we have these as default when initializing?

Reply:
Yes, I can do that for `dry_run` and`force`